src/ValueObject/NonEmptyString.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace ThePay\ApiClient\ValueObject;
+
+class NonEmptyString extends StringValue
+{
+    protected static function filter($value)
+    {
+        $string = parent::filter($value);
+
+        if (trim($string) === '') {
```

Comment:
Should be `' '` empty string, or not?

Reply:
IMHO it makes sense to use trim function, or just add a second parameter with default value `true` to switch this behavior.

Reply:
You mean something like `__construct($value, bool $throwOnWhiteCharsOnly = true)`?

```php
new NonEmptyString(' ', throwOnWhiteCharsOnly: false); // OK
new NonEmptyString(' ', throwOnWhiteCharsOnly: true); // throws
```

It will be problematic in next refactoring where the constructor became final.